components/coreapi/types.go

```diff
@@ -76,62 +87,86 @@ type infoResponse struct {
 	Name string `json:"name"`
 	// The server version of the node software.
 	Version string `json:"version"`
+	// The ID of the node
+	IssuerID string `json:"issuerId"`
 	// The current status of this node.
 	Status nodeStatus `json:"status"`
+	// The metrics of this node.
+	Metrics nodeMetrics `json:"metrics"`
 	// The protocol versions this node supports.
 	SupportedProtocolVersions protocol.Versions `json:"supportedProtocolVersions"`
 	// The protocol parameters used by this node.
-	ProtocolParameters *iotago.ProtocolParameters `json:"protocol"`
+	ProtocolParameters json.RawMessage `json:"protocol"`
 	// todo The base token of the network.
 	//BaseToken *protocfg.BaseToken `json:"baseToken"`
-	// The metrics of this node.
-	Metrics nodeMetrics `json:"metrics"`
 	// The features this node exposes.
 	Features []string `json:"features"`
 }
 
+//nolint:tagliatelle
 type nodeStatus struct {
 	// Whether the node is healthy.
 	IsHealthy bool `json:"isHealthy"`
+	// The blockID of last accepted block.
+	LastAcceptedBlockID string `json:"lastAcceptedBlockId"`
+	// The blockID of the last confirmed block
+	LastConfirmedBlockID string `json:"lastConfirmedBlockId"`
+	// The latest finalized slot.
+	FinalizedSlot iotago.SlotIndex `json:"latestFinalizedSlot"`
+	// The Accepted Tangle Time
+	ATT time.Time `json:"ATT"`
+	// The Relative Accepted Tangle Time
+	RATT time.Time `json:"RATT"`
+	// The Confirmed Tangle Time
+	CTT time.Time `json:"CTT"`
+	// The Relative Confirmed Tangle Time
+	RCTT time.Time `json:"RCTT"`
 	// The latest known committed slot info.
-	LatestCommittedSlot slotInfoResponse `json:"latestCommittedSlot"`
-	// The latest confirmed slot.
-	ConfirmedSlot slotInfoResponse `json:"latestConfirmedSlot"`
+	LatestCommittedSlot iotago.SlotIndex `json:"latestCommittedSlot"`
 	// The milestone index at which the last pruning commenced.
-	PruningIndex iotago.SlotIndex `json:"pruningIndex"`
+	PruningSlot iotago.SlotIndex `json:"pruningSlot"`
 }
 
 type nodeMetrics struct {
 	// The current rate of new blocks per second.
 	BlocksPerSecond float64 `json:"blocksPerSecond"`
-	// The current rate of referenced blocks per second.
-	ReferencedBlocksPerSecond float64 `json:"referencedBlocksPerSecond"`
-	// The ratio of referenced blocks in relation to new blocks of the last confirmed milestone.
-	ReferencedRate float64 `json:"referencedRate"`
+	// The current rate of confirmed blocks per second.
+	ConfirmedBlocksPerSecond float64 `json:"confirmedBlocksPerSecond"`
+	// The ratio of confirmed blocks in relation to new blocks of the last confirmed milestone.
+	ConfirmedRate float64 `json:"confirmedRate"`
 }
```

Comment:
Should this be "accepted" or "committed" rather than "confirmed"? Also, need to change the wording in relation to milestones. I don't really understand what the equivalent of ConfirmedRate would be without milestones or what the purpose of this metric is.

Reply:
Perhaps the names of these metrics are actually fine, but more description is needed in the comment to understand precisely what the metric actually tells you.